doc/source/data/training_ingest.rst

```diff
@@ -0,0 +1,525 @@
+        # Get an iterator to the dataset we passed in below.
+        it = session.get_dataset_shard("train")
+        for _ in range(2):
+            # Prefetch 10 batches at a time.
+            for batch in it.iter_batches(batch_size=128, prefetch_batches=10):
+                print("Do some training on batch", batch)
+
+    my_trainer = TorchTrainer(
+        train_loop_per_worker,
+        scaling_config=ScalingConfig(num_workers=2),
+        datasets={"train": ds},
+    )
+    my_trainer.fit()
+
+.. testoutput::
+    :hide:
+
+    ...
+
+Random Shuffling
+~~~~~~~~~~~~~~~~
+Randomly shuffling data for each epoch can be important for model quality depending on what model you are training.
+
+With Ray Data, there are 2 approaches to random shuffling:
+
+1. Shuffling data blocks & local shuffling on each training worker.
+2. Full global shuffle, which is more expensive.
+
+For most cases, option 1 will suffice. 
+
+First, randomize each :ref:`block <dataset_concept>` of your dataset via :meth:`randomize_block_order <ray.data.Dataset.randomize_block_order>`. Then, when iterating over your dataset during training, enable local shuffling by specifying a ``local_shuffle_buffer_size`` to :meth:`iter_batches <ray.data.DataIterator.iter_batches>` or :meth:`iter_torch_batches <ray.data.DataIterator.iter_torch_batches>`.
+
+.. testcode::
+    import ray
+
+     ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+
+    # Randomize the blocks of this dataset.
+    ds = ds.randomize_block_order()
+
+    def train_loop_per_worker():
+        # Get an iterator to the dataset we passed in below.
+        it = session.get_dataset_shard("train")
+        for _ in range(2):
+            # Use a shuffle buffer size 10k elements.
+            for batch in it.iter_batches(
+                local_shuffle_buffer_size=10000, batch_size=128):
+                print("Do some training on batch", batch)
+
+    my_trainer = TorchTrainer(
+        train_loop_per_worker,
+        scaling_config=ScalingConfig(num_workers=2),
+        datasets={"train": ds},
+    )
+    my_trainer.fit()
+
+.. testoutput::
+    :hide:
+
+    ...
+
+
+If your model is very dependent on shuffle quality, you can do a full global shuffle by calling :meth:`random_shuffle <ray.data.Dataset.random_shuffle>` on your dataset.
+
+.. testcode::
+
+    import ray
+
+     ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+
+    # Do a global shuffle of all rows in this dataset.
+    ds = ds.random_shuffle()
+
+For more information on how to optimize shuffling, and which approach to choose, see the :ref:`Optimize shuffling guide <optimizing_shuffles>`.
+
+Reproducibility
+~~~~~~~~~~~~~~~
+When developing or hyperparameter tuning models, it's important to have reproducibility during data ingest so that it does not affect model quality. Follow these 3 steps to enable reproducibility:
```

Comment:
```suggestion
When developing or hyperparameter tuning models, reproducibility is important during data ingest so that it does not affect model quality. Follow these 3 steps to enable reproducibility:
```

Reply:
Not sure if the second "it" is reproducibility, in which case my suggestion should work. If it's not, then, suggest replacing "it" with the correct noun.